### PR TITLE
Adds batching to queries made from cs

### PR DESCRIPTION
**Pull request by dposada**
## Changes proposed in this PR

- groups UUIDs into batches of at most 100 when querying
- modifies 2 integration tests (`show`, `wait`) to query for 200 jobs each

## Why are we making these changes?

We discovered a bug where querying for too many UUIDs in the same request was failing due to hitting the limit of the query string length. This change allows us to ask for as many UUIDs as needed, but in batches.

### CODE REVIEW

File: cli/cook/querying.py [hunk 1]
```diff
@@ -26,21 +26,16 @@ class Clusters:
 
 def __distinct(seq):
```
Comment (shamsimam) [inline cli/cook/querying.py]: Nitpick: should this be moved to `util.py`

File: cli/cook/querying.py [hunk 2]
```diff
@@ -72,6 +67,30 @@ def satisfy_pred():
     return entities
 
 
+def partition(l, n):
```
Comment (shamsimam) [inline cli/cook/querying.py]: Nitpick: should this be moved to `util.py`

File: cli/cook/querying.py [hunk 3]
```diff
@@ -72,6 +67,30 @@ def satisfy_pred():
...
+        return []
+
+    # Cook will give us back two copies if the user asks for the same UUID twice, e.g.
+    # $ cs show d38ea6bd-8a26-4ddf-8a93-5926fa2991ce d38ea6bd-8a26-4ddf-8a93-5926fa2991ce
+    # Prevent this by calling distinct:
+    uuids = __distinct(uuids)
+
+    entities = []
+    query_batch_size = 100
+    for uuid_batch in partition(uuids, query_batch_size):
```
Comment (shamsimam) [inline cli/cook/querying.py]: Should this be inside a try-catch block and the exceptions aggregated?

File: cli/cook/querying.py (hunk 1 above)
Comment (dposada) [reply cli/cook/querying.py]: Done

File: cli/cook/querying.py (hunk 2 above)
Comment (dposada) [reply cli/cook/querying.py]: Done

File: cli/cook/querying.py (hunk 3 above)
Comment (dposada) [reply cli/cook/querying.py]: `make_data_request` (which is called by `__query_cluster`) takes care of a couple types of exceptions:

```python
def make_data_request(cluster, make_request_fn):
    """
    Makes a request (using make_request_fn), parsing the
    assumed-to-be-JSON response and handling common errors
    """
    try:
        resp = make_request_fn()
        if resp.status_code == 200:
            return resp.json()
        else:
            return []
    except requests.exceptions.ConnectionError as ce:
        logging.exception(ce)
        raise Exception(f'Encountered connection error with {cluster["name"]} ({cluster["url"]}).')
    except requests.exceptions.ReadTimeout as rt:
        logging.exception(rt)
        raise Exception(f'Encountered read timeout with {cluster["name"]} ({cluster["url"]}).')
    except IOError as ioe:
        logging.exception(ioe)
        return []
    except json.decoder.JSONDecodeError as jde:
        logging.exception(jde)
        return []
```

Any others I think we actually want to bail out of the whole operation for.

**Comment by dposada** [comment]
@shamsimam RTM

**Pull request merged**